examples/ex-07.c

```diff
@@ -0,0 +1,772 @@
+
+   return 0;
+}
+
+int my_InitBasis(braid_App app, double t, int index, braid_Vector *u_ptr)
+{
+   my_Vector *u;
+
+   u = (my_Vector *)malloc(sizeof(my_Vector));
+
+   // initialize with the columns of the identity matrix
+   VecSet(u->values, 0.);
+   u->values[index] = 1.;
+
+   *u_ptr = u;
+
+   return 0;
+}
+
+int my_Clone(braid_App app, braid_Vector u, braid_Vector *v_ptr)
+{
+   my_Vector *v;
+
+   v = (my_Vector *)calloc(1, sizeof(my_Vector));
+   VecCopy(u->values, v->values);
+
+   *v_ptr = v;
+
+   return 0;
+}
+
+int my_Free(braid_App app, braid_Vector u)
+{
+   free(u);
+
+   return 0;
+}
+
+int my_Sum(braid_App app,
+           double alpha,
+           braid_Vector x,
+           double beta,
+           braid_Vector y)
+{
+   for (int i = 0; i < VecSize; i++)
+   {
+      y->values[i] = alpha * x->values[i] + beta * y->values[i];
+   }
+
+   return 0;
+}
+
+int my_InnerProd(braid_App app, braid_Vector u, braid_Vector v, double *prod_ptr)
+{
+   double dot = 0.;
+
+   for (int i = 0; i < VecSize; i++)
+   {
+      dot += (u->values[i]) * (v->values[i]);
+   }
+   *prod_ptr = dot;
+   return 0;
+}
+
+int my_SpatialNorm(braid_App app, braid_Vector u, double *norm_ptr)
+{
+   /* our inner product induces the Euclidean norm */
+   my_InnerProd(app, u, u, norm_ptr);
+   *norm_ptr = sqrt(*norm_ptr);
+   return 0;
+}
+
+int my_Access(braid_App app, braid_Vector u, braid_AccessStatus astatus)
+{
+   FILE *file = (app->file);
+   int index, i;
+   double t;
+
+   braid_AccessStatusGetT(astatus, &t);
+   braid_AccessStatusGetTIndex(astatus, &index);
+
+   fprintf(file, "%d", index);
+   for (i = 0; i < VecSize; i++)
+   {
+      fprintf(file, " %.14e", (u->values[i]));
+   }
+   fprintf(file, "\n");
+   fflush(file);
+
+   /* write the lyapunov vectors to file */
+   file = app->file_lv;
+   int local_rank, num_exp;
+   braid_AccessStatusGetDeltaRank(astatus, &local_rank);
+   num_exp = local_rank;
+   double *exponents = malloc(local_rank * sizeof(double));
+   braid_AccessStatusGetLocalLyapExponents(astatus, exponents, &num_exp);
+
+   fprintf(file, "%d", index);
+   for (int j = 0; j < local_rank; j++)
+   {
+      my_Vector *psi;
+      braid_AccessStatusGetBasisVec(astatus, &psi, j);
+      if (psi)
+      {
+         if (j < num_exp)
+         {
+            (app->lyap_exps)[j] += exponents[j];
+            fprintf(file, " %.14e", exponents[j]);
+         }
+         else
+         {
+            fprintf(file, " %.14e", 0.);
+         }
+         for (i = 0; i < VecSize; i++)
+         {
+            fprintf(file, " %.14e", (psi->values[i]));
+         }
+      }
+   }
+   fprintf(file, "\n ");
+   fflush(file);
+   free(exponents);
+
+   return 0;
+}
+
+int my_BufSize(braid_App app, int *size_ptr, braid_BufferStatus bstatus)
+{
+   *size_ptr = VecSize * sizeof(double);
+
+   /* tell braid the size of the basis vectors */
+   /* Note: this isn't necessary here, but for more complicated applications this size may be different */
```

Comment:
I would again rephrase this in terms of something new, the user must do in comparison to traditional Braid, so say something like,

In contrast to traditional braid, you can also specify the size of a single Lyapunov basis vector (in case it is different from that of the state vector).
Note: this isn't necessary here, but for more complicated applications this size may be different